src/Collection.js

```diff
@@ -118,6 +118,15 @@ class Collection {
     return this.__paths.length;
   }
 
+  /**
+   * Provides a callback mechanism to be notified when done with previous steps.
+   *
+   * @param {function} callback
+   */
+  done(cb) {
```

Comment:
Which use case does this solve? Why not do

```
Collection.fromNodes(...).forEach(...);
// do stuff afterwards
```

The `Collection` API is not asynchronous.